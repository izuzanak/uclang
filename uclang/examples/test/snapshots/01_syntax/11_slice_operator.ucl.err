 ---------------------------------------- 
Exception: ERROR: in file: "11_slice_operator.ucl" on line: 53
    ("Odd lines: %s\n" % $new File("10_slice_operator.ucl","r")[::2]).print();

Cannot open file "10_slice_operator.ucl"
 ---------------------------------------- 
