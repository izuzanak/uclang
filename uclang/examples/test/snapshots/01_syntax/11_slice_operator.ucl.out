
=== Array slice ================================================================

array: [1,2,3,4,5,6,7,8,9,10]
array[::2]: [1,3,5,7,9]
array[-1::-2]: [10,8,6,4,2]

=== String slice ===============================================================

"!!ddllrrooww oolllleeHH"[::-2]: Hello world!

=== Iter_0 slice ===============================================================

Iter_0[::]: [item(0),item(1),item(2),item(3),item(4),item(5),item(6),item(7),item(8),item(9)]
Iter_0[::-1]: [item(9),item(8),item(7),item(6),item(5),item(4),item(3),item(2),item(1),item(0)]

=== Iter_1 slice ===============================================================

Iter_1[::]: [item(0),item(1),item(2),item(3),item(4),item(5),item(6),item(7),item(8),item(9)]
Iter_1[::-1]: [item(9),item(8),item(7),item(6),item(5),item(4),item(3),item(2),item(1),item(0)]
Iter_1[4::]: [item(4),item(5),item(6),item(7),item(8),item(9)]
Iter_1[-4::-1]: [item(6),item(5),item(4),item(3),item(2),item(1),item(0)]

=== File lines slice ===========================================================

